src/legacy/core_plugins/kibana/public/doc/index.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import uiRoutes from 'ui/routes';
+// @ts-ignore
+import { getRootBreadcrumbs } from 'plugins/kibana/discover/breadcrumbs';
+// @ts-ignore
+import html from './index.html';
+import './doc_directive';
+import { IndexPatterns } from '../../../data/public/index_patterns';
+
+uiRoutes
+  // the old, pre 8.0 route, no longer used, keep it to stay compatible
+  // somebody might have bookmarked his favorite log messages
+  .when('/doc/:indexPattern/:index/:type', {
+    redirectTo: '/doc/:indexPattern/:index',
+  })
+  // the new route, es 7 deprecated types, es 8 removed them
+  .when('/doc/:indexPattern/:index', {
```

Comment:
Not really part of this PR, but why is there the name and the id of the pattern in the URL? Is this a performance optimization to not query the other indices? If it doesn't matter, it could be further simplified by just using the id of the index pattern and using the title of the saved object to do the actual lookup.

Reply:
Good question 👍 , yes, should be taken care of when we get rid of the angular router!

Reply:
there's an argument against that, if we use the index pattern title and it contains a wildcard, we're querying several indices. And there might be several documents with same id in different indices (With custom routing, there even id duplicate in one index in different shards)

Reply:
Ah, that makes sense, thanks.